rust/src/transformations/make_stable_expr/expr_sum/mod.rs

```diff
@@ -0,0 +1,347 @@
+
+        let pp_map = move |d_in: &IntDistance| match public_info {
+            MarginPub::Keys => TI::Sum::inf_cast(*d_in)?.inf_mul(&l.alerting_abs()?.total_max(u)?),
+            MarginPub::Lengths => TI::Sum::inf_cast(*d_in / 2)?.inf_mul(&u.inf_sub(&l)?),
+        };
+
+        Ok(StabilityMap::new_fallible(move |d_in: &MI::Distance| {
+            let (l0, l1, li) = MI::triple(d_in.clone());
+            let (l0, l1, li) = (TI::Sum::inf_cast(l0)?, pp_map(&l1)?, pp_map(&li)?);
+
+            let mcp = MI::max_changed_partitions(d_in.clone(), margin.max_num_partitions.clone())?;
+            let relaxation = TI::Sum::inf_cast(mcp)?.inf_mul(&pp_relaxation)?;
+
+            Ok(if P == 1 {
+                let ideal_sens = l1.total_min(l0.inf_mul(&li)?)?;
+                Scalar::from(ideal_sens.inf_add(&relaxation)?)
+            } else if P == 2 {
+                let (l0, l1, li) = (f64::inf_cast(l0)?, f64::inf_cast(l1)?, f64::inf_cast(li)?);
+                let relaxation = f64::inf_cast(relaxation)?;
+                let ideal_sens = l1.total_min(l0.inf_sqrt()?.inf_mul(&li)?)?;
+                Scalar::F64(ideal_sens.inf_add(&relaxation)?)
+            } else {
+                return fallible!(MakeTransformation, "unsupported Lp norm");
+            })
+        }))
+    }
+
+    let series_domain = input_domain.active_series()?.clone();
+    let stability_map = match dtype {
+        // DataType::UInt8 => stability_map::<MI, P, u8>(series_domain, margin),
+        // DataType::UInt16 => stability_map::<MI, P, u16>(series_domain, margin),
+        DataType::UInt32 => stability_map::<MI, P, u32>(series_domain, input_margin),
+        DataType::UInt64 => stability_map::<MI, P, u64>(series_domain, input_margin),
+        DataType::Int8 => stability_map::<MI, P, i8>(series_domain, input_margin),
+        DataType::Int16 => stability_map::<MI, P, i16>(series_domain, input_margin),
+        DataType::Int32 => stability_map::<MI, P, i32>(series_domain, input_margin),
+        DataType::Int64 => stability_map::<MI, P, i64>(series_domain, input_margin),
+        DataType::Float32 => stability_map::<MI, P, f32>(series_domain, input_margin),
+        DataType::Float64 => stability_map::<MI, P, f64>(series_domain, input_margin),
+        _ => fallible!(MakeTransformation, "unsupported data type"),
+    }?;
+
+    t_prior
+        >> Transformation::new(
+            input_domain,
+            output_domain,
+            Function::new_expr(Expr::sum),
+            input_metric.clone(),
+            LpDistance::default(),
+            stability_map,
+        )?
+}
+
+trait SumStabilityMap<MI: Metric, MO: Metric> {
+    fn make_stability_map(
+        margin: Margin,
+        input_metric: MI,
+        output_metric: MO,
+        max_size: usize,
+    ) -> Fallible<StabilityMap<MI, MO>>
+    where
+        (ExprDomain, MI): MetricSpace,
+        (ExprDomain, MO): MetricSpace;
+}
+
+// 1. spread into triple
```

Comment:
More explanation like this would be be great, but I'm really not sure what to do with this. Only section (3) is referenced, so I'm left wondering about the rest.